packages/react/src/components/numberInput/NumberInput.test.tsx

```diff
@@ -95,7 +103,36 @@ describe('<NumberInput /> spec', () => {
     await act(async () => {
       userEvent.click(screen.getByRole('button', { name: 'Reset number input' }));
     });
-    const numberInput = screen.getByLabelText('Test label number input', { selector: 'input' });
+    const numberInput = getNumberInputElement();
     expect(numberInput).toHaveValue(null); // Empty string is converted to null by react/html
   });
+
+  it('should notify screen reader when user clicks step down or up button', async () => {
+    const result = render(<NumberInput defaultValue={10} step={10} {...numberInputProps} />);
+    userEvent.click(screen.getByRole('button', { name: numberInputProps.minusStepButtonAriaLabel }));
+    expect(getAriaLiveElement(result)).toHaveTextContent('0');
+    userEvent.click(screen.getByRole('button', { name: numberInputProps.plusStepButtonAriaLabel }));
+    expect(getAriaLiveElement(result)).toHaveTextContent('10');
+  });
+
+  it('aria-live element is not rendered unless button is pressed and is removed when input changes without buttons.', async () => {
+    const result = render(<NumberInput defaultValue={10} step={10} {...numberInputProps} />);
+    expect(getAriaLiveElement(result)).toBeNull();
+    userEvent.click(screen.getByRole('button', { name: numberInputProps.minusStepButtonAriaLabel }));
+    expect(getAriaLiveElement(result)).not.toBeNull();
+    await act(async () => {
+      userEvent.type(getNumberInputElement(), '20');
+    });
+    expect(getAriaLiveElement(result)).toBeNull();
+  });
+
+  it('input and buttons are grouped when component has the step-attribute', async () => {
+    const resultWithStep = render(<NumberInput defaultValue={10} step={10} {...numberInputProps} />);
+    const group = resultWithStep.container.querySelector('[role="group"]');
+    expect(group).not.toBeNull();
+
+    const resultWithoutStep = render(<NumberInput defaultValue={10} {...numberInputProps} />);
+    const groupNotFound = resultWithoutStep.container.querySelector('[role="group"]');
+    expect(groupNotFound).toBeNull();
+  });
```

Comment:
Nice accessibility tests. I'm just not sure if the last one about checking for role="group" is a bit too implementation detailed test? If we or unicus decides that actually role="group" isn't good enough and we have to change them, then this will fail. Although that might not happen, at least for a good while. What do you think?

Reply:
Role="group" is required in this case, so it should not be removed and the test should therefore be reliable. The element where the role-attribute is added, exists in all cases (just without the group attribute), so adding a data-testid would not work